### PR TITLE
CFY-4671 added tests for roles reload functionality

### PR DESCRIPTION
**Pull request by adamlavie**

### CODE REVIEW

File: rest-service/manager_rest/test/security/security_test_base.py [hunk 1]
```diff
@@ -139,10 +139,13 @@ def get_userstore_configuration(self):
...
 
     @staticmethod
-    def get_authorization_provider_configuration():
+    def get_roles_config_file_path():
         abs_path = os.path.dirname(os.path.abspath(__file__))
-        roles_config_file_path = os.path.join(abs_path,
-                                              '../resources/roles_config.yaml')
+        return os.path.join(abs_path, '../resources/roles_config.yaml')
+
+    @staticmethod
```
Comment (adamlavie) [inline rest-service/manager_rest/test/security/security_test_base.py]: @classmethod

File: rest-service/manager_rest/test/security/test_userstore_and_roles_reload.py [hunk 2]
```diff
@@ -28,12 +28,17 @@
...
+ROLES_CONFIG_FILE_INVALID_DICT = 'Failed parsing'
+ROLES_CONFIG_LOADED_SUCCESSFULLY = 'Loading of roles configuration ended ' \
+                                   'successfully'
+
 
 @attr(client_min_version=2, client_max_version=base_test.LATEST_API_VERSION)
 class TestUserstoreReloadFile(SecurityTestBase):
 
     def setUp(self):
         self.modify_userstore_file(self.get_users(), self.get_groups())
```
Comment (adamlavie) [inline rest-service/manager_rest/test/security/test_userstore_and_roles_reload.py]: wrap with restore func

File: rest-service/manager_rest/test/security/test_userstore_and_roles_reload.py [hunk 3]
```diff
@@ -47,14 +52,27 @@ def _init_test(self):
         self.modify_userstore_file(self.get_users(),
                                    self.get_groups(),
                                    USERSTORE_LOADED_SUCCESSFULLY)
+        self.restore_roles_config(ROLES_CONFIG_LOADED_SUCCESSFULLY)
+        # return self._create_secured_client('alice', 'alice_password')
```
Comment (adamlavie) [inline rest-service/manager_rest/test/security/test_userstore_and_roles_reload.py]: remove

File: rest-service/manager_rest/test/security/test_userstore_and_roles_reload.py [hunk 4]
```diff
@@ -47,14 +52,27 @@ def _init_test(self):
...
-            create_auth_header(username='alice',
-                               password='alice_password')
+            create_auth_header(username=username,
+                               password=password)
         client = self.create_client(headers=auth_header)
         return client
 
+    def restore_roles_config(self, message=None):
+        with open(self.get_roles_config_file_path(), 'w') as outfile:
+            outfile.write(yaml.dump(self.get_valid_roles_config(),
```
Comment (adamlavie) [inline rest-service/manager_rest/test/security/test_userstore_and_roles_reload.py]: dump ++> safe_dump

File: rest-service/manager_rest/test/security/test_userstore_and_roles_reload.py [hunk 5]
```diff
@@ -47,14 +52,27 @@ def _init_test(self):
         self.modify_userstore_file(self.get_users(),
                                    self.get_groups(),
                                    USERSTORE_LOADED_SUCCESSFULLY)
+        self.restore_roles_config(ROLES_CONFIG_LOADED_SUCCESSFULLY)
+        # return self._create_secured_client('alice', 'alice_password')
+
+    def _create_secured_client(self, username, password):
```
Comment (adamlavie) [inline rest-service/manager_rest/test/security/test_userstore_and_roles_reload.py]: remove user/pass

File: rest-service/manager_rest/test/resources/roles_config.yaml [hunk 6]
```diff
@@ -1,33 +1,2 @@
-# This file maps roles to allowed and denied permissions.
```
Comment (dankilman) [inline rest-service/manager_rest/test/resources/roles_config.yaml]: restore original file

File: rest-service/dev-requirements.txt [hunk 7]
```diff
@@ -1,6 +1,6 @@
 https://github.com/cloudify-cosmo/cloudify-dsl-parser/archive/3.3.1.zip
 https://github.com/cloudify-cosmo/cloudify-rest-client/archive/3.3.1.zip
-https://github.com/cloudify-cosmo/flask-securest/archive/0.8.zip
+https://github.com/cloudify-cosmo/flask-securest/archive/CFY-4671-reload-roles-on-modification.zip
```
Comment (dankilman) [inline rest-service/dev-requirements.txt]: don't forget to restore branch

**Pull request merged**